core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -105,15 +105,22 @@ public FateExecutor(Fate<T> fate, T environment, Set<Fate.FateOperation> fateOps
   protected void resizeFateExecutor(Map<Set<Fate.FateOperation>,Integer> poolConfigs,
       long idleCheckIntervalMillis) {
     final var pool = transactionExecutor;
-    final var runningTxRunners = getRunningTxRunners();
+    final var runningTxRunnersCopy = getRunningTxRunners();
     final int configured = poolConfigs.get(fateOps);
     ThreadPools.resizePool(pool, () -> configured, poolName);
-    final int needed = configured - runningTxRunners.size();
+    final int needed = configured - runningTxRunnersCopy.size();
     if (needed > 0) {
       // If the pool grew, then ensure that there is a TransactionRunner for each thread
       for (int i = 0; i < needed; i++) {
         try {
-          pool.execute(new TransactionRunner());
+          final TransactionRunner tr = new TransactionRunner();
+          synchronized (runningTxRunners) {
+            if (pool.isShutdown() || pool.isTerminating()) {
+              return;
+            }
+            runningTxRunners.add(tr);
+            pool.execute(tr);
+          }
         } catch (RejectedExecutionException e) {
           // RejectedExecutionException could be shutting down
           if (pool.isShutdown()) {
```

Comment:
I was thinking this case should probably be removed, but it might still be possible if the pool is shutdown after the `if (pool.isShutdown() || pool.isTerminating())` check but before `pool.execute(tr);`... So I think we still need it